components/fl-operator/Makefile

```diff
@@ -37,38 +39,44 @@ help: ## Display this help.
 
 ##@ Development
 
+bin/controller-gen:
+	GOBIN="$(GOBIN)" go install sigs.k8s.io/controller-tools/cmd/controller-gen
+
+bin/setup-envtest:
+	GOBIN="$(GOBIN)" go install sigs.k8s.io/controller-runtime/tools/setup-envtest
```

Comment:
The reason of not installing these only locally in the project, and instead in the `Dockerfile` was to reduce the possible duplicate downloads of the `controller-gen` and `setup-envtest`. I would prefer to reduce that duplication in the Dockerfile and/or rely that these exist in the OS.

Reply:
I have a different point-of-view: These tools are closely tied to the `controller-runtime` dependency. I.e. older versions of `controller-gen` will fail with new versions of `controller-runtime` and vice versa. Therefore it is important to ensure that the right version of these tools is in use and that we don't rely on existing installations of this tools in the system `$PATH`. Which makes it essential to track the versions of these tools as part of the project -- ideally in `go.mod` as introduced in this PR -- and install them locally only for this project.

Reply:
I see. I was not aware that the `controller-runtime` versions were not backwards-compatible. Then it makes sense to have them as you propose.

Reply:
And keep in mind, as I'm not sure if that's easily visible from the Makefile, that the binaries are not installed globally but only for the project by overriding `GOBIN` and `PATH` when these tools are needed.